test/actor_test.py

```diff
@@ -1333,50 +1333,43 @@ def inc(self):
             self.x += 1
             return self.x
 
-    local_plasma = ray.worker.global_worker.plasma_client.store_socket_name
-
     # Create an actor that is not on the local scheduler.
     actor = Counter.remote()
-    while ray.get(actor.local_plasma.remote()) == local_plasma:
+    while (ray.get(actor.local_plasma.remote()) !=
+           remote_node.get_plasma_store_name()):
         actor = Counter.remote()
 
     ids = [actor.inc.remote() for _ in range(100)]
 
     # Wait for the last task to finish running.
     ray.get(ids[-1])
 
-    # Kill the second plasma store to get rid of the cached objects and
-    # trigger the corresponding local scheduler to exit.
-    process = ray.services.all_processes[
-        ray.services.PROCESS_TYPE_PLASMA_STORE][1]
-    process.kill()
-    process.wait()
+    # Kill the second node.
+    head_node_cluster.remove_node(remote_node)
```

Comment:
I would kill another node and repeat the `ray.get` after this to make sure that we hit the `max_reconstructions=1` and `remaining_reconstructions=0` case.

Reply:
Actually, it looks like the test in `multi_node_test_2.py` does what I'm requesting. Can we deduplicate the tests?

Reply:
Removed this task and copied the `multi_node_test_2` one to this file and renamed it to `test_actor_reconstruction_on_node_failure`